crypto/s2n_certificate.h

```diff
@@ -19,7 +19,22 @@
 
 #include "crypto/s2n_rsa.h"
 #include "stuffer/s2n_stuffer.h"
-#include "tls/s2n_config.h"
+
+#define S2N_MAX_SERVER_NAME 256
+
+struct s2n_cert_chain {
+    struct s2n_blob cert;
+    struct s2n_cert_chain *next;
+};
+
+struct s2n_cert_chain_and_key {
+    uint32_t chain_size;
+    struct s2n_cert_chain *head;
+    struct s2n_rsa_private_key private_key;
```

Comment:
I'm not sure about the way we're modeling the certificate/chain/key here. What does it mean to have a certificate chain object that contains a cert and has a pointer to another certificate chain? Why does s2n_cert_chain_and_key have a single private key associated with it? Does oscp_status pertain to the entire chain or just an individual cert?

To me, a more logical model, would be to have an s2n_cert struct that contains a private key (and/or public key) and a pointer to the next s2n_cert. I don't know of any properties that you would need to associate specifically with an "s2n_cert_chain". Most properties would be pertinent to the individual certs (except for maybe chain length), so you could likely do away with a cert chain struct. 

I saw that this was just a copy and paste from s2n_config, and agree that these belong better in this file. I think it's out of the scope of this review, but I wanted to bring it up in case the changes in the client auth PRs would be touching this code a lot. I'd love to be able to revisit this if we don't address it with these PRs.